src/Symfony/Component/Validator/Tests/Constraints/BackedEnumValueValidatorTest.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Tests\Constraints;
+
+use Symfony\Component\Validator\Constraints\BackedEnumValue;
+use Symfony\Component\Validator\Constraints\BackedEnumValueValidator;
+use Symfony\Component\Validator\Exception\ConstraintDefinitionException;
+use Symfony\Component\Validator\Test\ConstraintValidatorTestCase;
+
+/**
+ * @author Aurélien Pillevesse <aurelienpillevesse@hotmail.fr>
+ */
+class BackedEnumValueValidatorTest extends ConstraintValidatorTestCase
+{
+    protected function createValidator(): BackedEnumValueValidator
+    {
+        return new BackedEnumValueValidator();
+    }
+
+    public function testExpectEnumForTypeAttribute()
+    {
+        $this->expectException(ConstraintDefinitionException::class);
+        $constraint = new BackedEnumValue(
+            type: self::class
+        );
+    }
+
+    public function testNullIsValid()
+    {
+        $this->validator->validate(
+            null,
+            new BackedEnumValue(
+                type: MyStringEnum::class
+            )
+        );
+
+        $this->assertNoViolation();
+    }
+
+    public function testEmptyStringIsValid()
+    {
+        $this->validator->validate(
+            '',
+            new BackedEnumValue(
+                type: MyStringEnum::class
+            )
+        );
+
+        $this->assertNoViolation();
+    }
+
+    public function testStringEnumValid()
+    {
+        $this->validator->validate(
+            'yes',
+            new BackedEnumValue(
+                type: MyStringEnum::class
```

Comment:
This should not reference non-autoloadable enums defined in a different test file as this would break if the other test file has not been loaded